transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -33,9 +33,10 @@ def fp8_gemm(
     use_bias: bool = False,
     use_split_accumulator: bool = False,
     D_dtype: Optional[tex.DType] = None,
-    ub_algo: tex.UbufOverlapAlgo = None,
-    ub: Union[tex.UbufCommOverlap, tex.UbufP2PCommOverlap] = None,
+    ub_algo: tex.NVTE_Comm_Overlap_Algo = None,
+    ub: Union[tex.CommOverlap, tex.CommOverlapP2P] = None,
     extra_output_tensor: torch.Tensor = None,
+    bulk_ubuf_fp8_type: Optional[tex.DType] = None,
```

Comment:
While we're changing the API, it may worth considering putting UB-specific options in `dict`s:
```suggestion
    extra_output_tensor: torch.Tensor = None,
    ub_options: Optional[Dict[str, Any]] = None,
```
UB's API is unstable, and this puts a burden on downstream users (see [versioning logic in Mcore](https://github.com/NVIDIA/NeMo/blob/a7fbf6babe6d8a2c9f53b9e72f260151833297c1/nemo/collections/nlp/modules/common/megatron/transformer.py#L840-L860)). By wrapping everything in a `dict`, TE can take more responsibility for backward compatibility, i.e. if a user provides a `dict` with options from an older version.

That said, these functions are considered internal interfaces. This is more of a concern with external APIs like the modules.

Reply:
I'm planning to implement this when we have to change all the UB-specific kwarg names during the cuBlasMp integration. The dict will help us support the old kwargs with a deprecation warning, without breaking downstream dependencies.